icu4c/source/i18n/number_longnames.cpp

```diff
@@ -169,6 +205,23 @@ void getMeasureData(const Locale &locale, const MeasureUnit &unit, const UNumber
     key.append("/", status);
     key.append(subtypeForResource, status);
 
+    // Grab desired case first, if available. Then grab nominative case to fill
+    // in the gaps.
+    //
+    // TODO(icu-units#138): implement spec-compliant fallback: case should
+    // fallback to "nominative" before falling back to no case. For the time
+    // being: as of CLDR 39 explicit "nominative" is not yet in use.
+    if (width == UNUM_UNIT_WIDTH_FULL_NAME && !unitDisplayCase.empty() &&
+        unitDisplayCase.compare("nominative") != 0) {
```

Comment:
shall we just process the nominative, even the process is doing nothing

Reply:
True! Thanks for the catch. That would even be more spec compliant. Here is the specification of the fallback process I should have implemented:
https://www.unicode.org/reports/tr35/tr35.html#Lateral_Inheritance
It should fall back to "nominative" before it falls back to no-case.

Reply:
Done.